source/js/components/petition/petition.jsx

```diff
@@ -116,26 +117,24 @@ export default class Petition extends React.Component {
     );
   }
 
-  fbButtonClicked() {
-    this.shareButtonClicked(`#share-progress-fb`);
-  }
-
-  twButtonClicked() {
-    this.shareButtonClicked(`#share-progress-tw`);
-  }
-
-  emButtonClicked() {
-    this.shareButtonClicked(`#share-progress-em`);
-  }
-
-  shareButtonClicked(id) {
+  shareButtonClicked(event, shareProgressButtonId) {
     ReactGA.event({
       category: `petition`,
       action: `share tap`,
       label: `${document.title} - share tap`
     });
 
-    document.querySelector(id + ` a`).click();
+    if (shareProgressButtonId) {
+      let shareProgressButton = document.querySelector(
+        `#${shareProgressButtonId} a`
+      );
+
+      if (shareProgressButton) {
+        document.querySelector(`#${shareProgressButtonId} a`).click();
```

Comment:
What if we removed L132-L134 and instead just did `shareProgressButton.click()`?

Reply:
Yea line 133 can def be changed to `shareProgressButton.click()` but we still need to keep the if statement just in case `shareProgressButton` is null.

Reply:
why am i just seeing this??? smh github